runtime/compiler/control/CompilationThread.cpp

```diff
@@ -3873,6 +3900,7 @@ TR::CompilationInfoPerThread::doSuspend()
    _compInfo.setSuspendThreadDueToLowPhysicalMemory(false);
    getCompThreadMonitor()->enter();
    setCompilationThreadState(COMPTHREAD_SUSPENDED);
+   _compInfo.getCompilationMonitor()->notifyAll();
```

Comment:
@mpirvu this is the only part of this change I'm unsure about. Should I be creating another state, something like `COMPTHREAD_SIGNAL_CHECKPOINT` and then only notifyAll if we're suspending due to a checkpoint request?

Reply:
Specifically, I don't know if this is ok to do if we're suspending for the usual (non-checkpoint) reasons.

Reply:
I don't find any flaws yet, though you are breaking new ground. Waiting on the compMonitor was only done by the compilation threads waiting for work and an application thread was supposed to signal them. You have the exact opposite here: app thread waiting and comp thread notifying. I am not sure whether or not there are some imcplicit assumptions in the existing code.